standalone/WebSocketExportToOrg.py

```diff
@@ -0,0 +1,109 @@
+"""
+This script export WebSocket communication to Emacs Org File.
+Script require the external Module PyOrgMode (https://github.com/bjonnh/PyOrgMode)
+To add an external module: Tools -> Options -> Jython 
+
+Export Format:
+
+* (#<ChannelID>)<WebSocket Channel host name>
+** Switching Protocol
+*** Handshake Request
+	#+BEGIN_SRC html
+	<Http Handshake Request>
+	#+END_SRC
+*** Handshake Respone
+	#+BEGIN_SRC html
+	<Http Handshake Request>
+	#+END_SRC
+** Messages
+*** (#<ChannelID>.<MessageID) <Direction> <Opcode> <Date Time> <Byte Lenght>
+	- Text
+	<Text Payload>
+	- Hex
+	<Hex Payload>
+"""
+
+from org.parosproxy.paros.control import Control
+from org.parosproxy.paros.extension.history import ExtensionHistory
+from org.zaproxy.zap.extension.websocket import ExtensionWebSocket
+from org.zaproxy.zap.extension.websocket import WebSocketChannelDTO
+from org.zaproxy.zap.extension.websocket import WebSocketMessageDTO
+import PyOrgMode
+import sys
+import binascii
+
+reload(sys)
+sys.setdefaultencoding('utf-8')
+
+extWebSocket = Control.getSingleton().getExtensionLoader().getExtension(ExtensionWebSocket.NAME)
+extHist = Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.NAME)
+
+base = PyOrgMode.OrgDataStructure()
+channelId = 0
+if(extHist != None):
+  i=1
+  # Loop through the history table, printing out the history id and the URL
+  hr = extHist.getHistoryReference(i)
+  while(hr != None):
```

Comment:
We can have null HistoryRecords without them being at the end.
https://github.com/zaproxy/community-scripts/blob/master/standalone/Loop%20through%20history%20table.js#L11-L13 is the best way to check for the last record

Reply:
It could also obtain the channels directly (e.g. `extWebSocket.getStorage().getTable().getChannelItems()`) instead of recreating them based on the HTTP message (the handshake can be obtained from the channel, if it still exists).

Reply:
@thc202 That will work also with closed connections?